tests/test_prior.py

```diff
@@ -46,6 +49,41 @@ def test_log_prior_with_no_prior():
     assert param.log_prior().numpy() == 0.0
 
 
+def test_log_prior_for_uniform_prior():
+    """
+    A parameter with a uniform prior should have uniform log-prior,
+    even if it has a transform to constrain it.
+    """
+
+    uniform_prior = Uniform(low=np.float64(0), high=np.float64(100))
+    param = gpflow.Parameter(1., transform=gpflow.utilities.positive(),
+                             prior=uniform_prior)
+    low_value = param.log_prior().numpy()
+    param.assign(10.)
+    high_value = param.log_prior().numpy()
+
+    assert np.isclose(low_value, high_value)
+
+
+def test_log_prior_on_unconstrained():
+    """
+    A parameter with an Exp transform, and a uniform prior on its unconstrained, should have a
+    prior which scales as  1/value.
```

Comment:
```suggestion
    prior in the constrained space that scales as 1/value.
```
is this what you mean?